Library/Tracking/TrackingClientConfiguration.swift

```diff
@@ -0,0 +1,132 @@
+import Foundation
+import KsApi
+
+public enum TrackingClientIdentifier: String {
+  case dataLake = "DataLake"
+  case koala = "Koala"
+
+  var emoji: String {
+    switch self {
+    case .dataLake: return "💧"
+    case .koala: return "🐨"
+    }
+  }
+}
+
+public typealias TrackingClientRecordDictionary = (String, [String: Any]) -> [String: Any]
+public typealias TrackingClientRequest = (TrackingClientConfiguration, EnvironmentType, Data) -> URLRequest?
+public typealias TrackingClientURL = (EnvironmentType) -> URL?
+
+public struct TrackingClientConfiguration {
+  public let envelope: (Any) -> Any
+  public let httpMethod: KsApi.Method
+  public let identifier: TrackingClientIdentifier
+  public let recordDictionary: TrackingClientRecordDictionary
+  public let request: TrackingClientRequest
+  public let url: TrackingClientURL
+}
+
+extension TrackingClientConfiguration {
+  public static let koala: TrackingClientConfiguration = .init(
+    envelope: { $0 },
+    httpMethod: .POST,
+    identifier: .koala,
+    recordDictionary: koalaRecordDictionary,
+    request: koalaRequest,
+    url: koalaUrl
+  )
+
+  public static let dataLake: TrackingClientConfiguration = .init(
+    envelope: { records in ["records": records] },
+    httpMethod: .PUT,
+    identifier: .dataLake,
+    recordDictionary: dataLakeRecordDictionary,
+    request: dataLakeRequest,
+    url: dataLakeUrl
+  )
+}
+
+// MARK: - Koala Functions
+
+private let koalaRecordDictionary: TrackingClientRecordDictionary = { event, properties in
+  ["event": event, "properties": properties]
+}
+
+private let koalaRequest: TrackingClientRequest = { config, environmentType, data in
+  guard
+    let baseUrl = config.url(environmentType),
+    var components = URLComponents(url: baseUrl, resolvingAgainstBaseURL: false)
+  else { return nil }
+
+  let dataString = data.base64EncodedString(options: [])
+
+  if dataString.count >= 10_000 {
+    // swiftlint:disable:next line_length
+    print("\(config.identifier.emoji) [\(config.identifier) Error]: Base64 payload is longer than 10,000 characters.")
```

Comment:
This was just something we printed at this point previously, not sure how important it is to keep.